test/unit/x20schemapack.test.js

```diff
@@ -0,0 +1,230 @@
+      input = {
+        type: 'file',
+        data: fileSource
+      },
+      schemapack = new SchemaPack(input);
+    expect(schemapack);
+  });
+});
+
+describe('getMetaData method', function() {
+  it('Should return the provided input metadata', function() {
+    const fileSource = path.join(__dirname, SWAGGER_20_FOLDER_JSON + 'sampleswagger.json'),
+      input = {
+        type: 'file',
+        data: fileSource
+      },
+      schemapack = new SchemaPack(input);
+    schemapack.getMetaData((error, result) => {
+      expect(error).to.be.null;
+      expect(result.result).to.be.true;
+      expect(result.name).to.be.equal('Swagger Petstore');
+    });
+  });
+});
+
+describe('Convert method', function() {
+  it('Should convert an example file from: ', function(done) {
+    const fileSource = path.join(__dirname, SWAGGER_20_FOLDER_JSON, 'sampleswagger.json'),
+      fileData = fs.readFileSync(fileSource, 'utf8'),
+      input = {
+        type: 'string',
+        data: fileData
+      },
+      schemapack = new SchemaPack(input);
+    schemapack.convert((error, result) => {
+      expect(error).to.be.null;
+      expect(result.result).to.be.true;
+    });
+    done();
+  });
+});
+
+describe('Swagger 2.0  schemapack mergeAndValidate method', function() {
+  it('Should merge correctly the files in folder - petstore separated', function(done) {
+    let folderPath = path.join(__dirname, '../data/swaggerMultifile/petstore-separate-yaml'),
+      files = [],
+      array = [
+        { fileName: folderPath + '/spec/swagger.yaml' },
+        { fileName: folderPath + '/spec/Pet.yaml' },
+        { fileName: folderPath + '/spec/parameters.yaml' },
+        { fileName: folderPath + '/spec/NewPet.yaml' },
+        { fileName: folderPath + '/common/Error.yaml' }
+      ];
+
+    array.forEach((item) => {
+      files.push({
+        content: fs.readFileSync(item.fileName, 'utf8'),
+        fileName: item.fileName
+      });
+    });
+
+    var schema = new SchemaPack({ type: 'folder', data: files });
+    schema.mergeAndValidate((err, status) => {
+      if (err) {
+        expect.fail(null, null, err);
+      }
+      if (status.result) {
+        schema.convert((error, result) => {
+          if (error) {
+            expect.fail(null, null, err);
+          }
+          expect(result.result).to.equal(true);
+          expect(result.output.length).to.equal(1);
+          expect(result.output[0].type).to.have.equal('collection');
+          expect(result.output[0].data).to.have.property('info');
+          expect(result.output[0].data).to.have.property('item');
+          done();
+        });
+      }
+      else {
+        expect.fail(null, null, status.reason);
+        done();
+      }
+    });
+  });
+
+  it('Should merge correctly the files in folder - basicExample', function(done) {
+    let folderPath = path.join(__dirname, '../data/swaggerMultifile/uberTest'),
```

Comment:
We are using `uberTest` API definition here instead of `basicExample` the one that's mentioned in the title.